engine/hooks/scheduler.go

```diff
@@ -162,6 +162,7 @@ func (s *Service) deleteTaskExecutionsRoutine(c context.Context) error {
 				continue
 			}
 			for _, t := range tasks {
+				branchDeletionInProgress := 0
```

Comment:
Instead of adding this kind of counter, maybe we can add a flag task to delete and set it to true when we remove the task execution.